core/controllers/access_validators_test.py

```diff
@@ -416,3 +418,59 @@ def test_validation_returns_false_if_given_user_is_non_existent(
             ), expected_status_int=404
         )
         self.logout()
+
+
+class TopicEditorPageAccessValidationHandlerTests(test_utils.EmailTestBase):
+    """Checks the access to the topic editor page and its rendering."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.add_user_role(
+            self.CURRICULUM_ADMIN_USERNAME, feconf.ROLE_ID_CURRICULUM_ADMIN)
+
+        self.topic_id = topic_fetchers.get_new_topic_id()
```

Comment:
It looks like you are just fetching a new ID but not creating a new topic because this `if topic is None:` is working and `PageNotFoundException` is raised!

Reply:
@hrshkshri I think this will resolve the 404 issue!

Reply:
Hmmm, right.